File: javascript/matrix/matrix.js
// Exercism Matrix
export class Matrix{
  constructor(matrix){
    this.rows = matrix
        .split('\n')
        .map(i => i.split(' ').map(Number));
    this.columns = this.transpose();
    }
  
    transpose() {
      return Object.keys(this.rows[0]).map(i => {
        return this.rows.map(j => j[i]);
      });
    }
}

File: ruby/matrix/matrix.rb
# frozen_string_literal: true

# Matrix exercism
class Matrix
  attr_reader :matrix_string

  def initialize(matrix_string)
    @matrix_string = matrix_string
  end

  def rows
    @rows ||= matrix_string.each_line.map { |key| key.split.map(&:to_i) }
  end

  def columns
    rows.transpose
  end
end
